provider/datagen/tests/collator-extended-data-test.rs

```diff
@@ -0,0 +1,125 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::cmp::Ordering;
+
+use icu_collator::provider::CollationDataV1Marker;
+use icu_collator::{Collator, CollatorOptions, Strength};
+use icu_datagen::CldrLocaleSubset;
+use icu_datagen::SourceData;
+use icu_locid::locale;
+use icu_locid::{langid, Locale};
+use icu_provider::AsDowncastingAnyProvider;
+use icu_provider::AsDynProviderAnyMarkerWrap;
+use icu_provider::{AnyMarker, DynProvider};
+use icu_provider_adapters::fallback::{LocaleFallbackProvider};
+use lazy_static::lazy_static;
+use icu_provider::*;
+
+lazy_static! {
+    static ref SOURCE_DATA: SourceData = SourceData::default()
+        .with_icuexport(icu_testdata::paths::icuexport_toml_root())
+        .unwrap()
+        // CLDR data is needed for locale fallback support
+        .with_cldr(icu_testdata::paths::cldr_json_root(), CldrLocaleSubset::Full)
+        .unwrap();
+}
+
+fn get_provider() -> impl DynProvider<AnyMarker> {
```

Comment:
> Can we run this on `NormalizationProvider`?

I need to fork between `NormalizationProvider` and `CollationProvider` but yes.

> It might be a bit less efficient, but it avoids the indirection through testdata/postcard, and is also how other transformers are tested.

Don't know what you mean by this. I'm not using postcard in this test. I am pointing the datagen provider (which forks into NormalizationProvider and CollationProvider) to the raw data that lives in testdata.

Reply:
I left this as the full provider for now.